internal/api/controllers/private/highlevelJobRequest.go

```diff
@@ -0,0 +1,159 @@
+package private
+
+import (
+	"net/http"
+	"playbook-dispatcher/internal/api/controllers/public"
+	"playbook-dispatcher/internal/api/middleware"
+	"playbook-dispatcher/internal/common/model/generic"
+	"playbook-dispatcher/internal/common/utils"
+
+	"github.com/google/uuid"
+	"github.com/labstack/echo/v4"
+)
+
+func (this *controllers) ApiInternalHighlevelJobRequest(ctx echo.Context) error {
+	var input JobRequestBody
+	var response []JobRequestInfo
+
+	err := utils.ReadRequestBody(ctx, &input)
+	if err != nil {
+		utils.GetLogFromEcho(ctx).Error(err)
+		return ctx.NoContent(http.StatusBadRequest)
+	}
+
+	hostConnectorDetails, err := this.inventoryConnectorClient.GetHostConnectionDetails(
+		ctx.Request().Context(),
+		input.Hosts,
+		this.config.GetString("inventory.connector.ordered.how"),
+		this.config.GetString("inventory.connector.ordered.by"),
+		this.config.GetInt("inventory.connector.limit"),
+		this.config.GetInt("inventory.connector.offset"),
+	)
+	if err != nil {
+		utils.GetLogFromEcho(ctx).Error(err)
+		return ctx.NoContent(http.StatusBadRequest)
+	}
+
+	satellite, directConnected, noRhc := sortHostsByRecipient(hostConnectorDetails)
+
+	runInputs := []generic.RunInput{}
+	runInputConnInfo := []RecipientWithConnectionInfo{}
+
+	// when RHC is not setup, do not dispatch playbook run request
+	if noRhc != nil {
+		noRHCRecipientConnInfo := []RecipientWithConnectionInfo{getRHCStatus(noRhc, input.OrgId)}
+		for _, noRHCRecipientConn := range noRHCRecipientConnInfo {
+			response = append(response, JobRequestInfo{
+				Recipient:       noRHCRecipientConn.Recipient,
+				Status:          noRHCRecipientConn.Status,
+				RequestDispatch: "failure",
+				Systems:         noRHCRecipientConn.Systems,
+			})
+		}
+	}
```

Comment:
When RHC is not setup, we are not dispatching playbook run request. Wanted to verify if this is accurate. Any other special thing we should do here?